src/controller/c/supervisor.c

```diff
@@ -387,6 +387,9 @@ static WbNodeRef self_node_ref = NULL;
 static WbNodeRef position_node_ref = NULL;
 static WbNodeRef export_string_node_ref = NULL;
 static WbNodeRef orientation_node_ref = NULL;
+static double *relative_pose = NULL;
```

Comment:
Instead of dynamic allocation / free, it would be simpler and faster to have a stack allocation, isn't it?
```suggestion
static double relative_pose[16];
```

Reply:
Hmm, that is a good point, but it is strange that there are a lot of variables doing the opposite like `solid_velocity`, `virtual_reality_headset_position`, `virtual_reality_headset_orientation`... I am not sure about the reason, maybe we wanted to make sure the memory chunk is invalid after a new value is requested?

Reply:
I don't see a good reason for that.
I believe we should also simplify the other cases (`solid_velocity`, etc.) in a different PR.

Reply:
I don't remember exactly, but one reason is that a pointer also contains one more information than an array, i.e. if the value is set or not.
For example if some errors occurs and the value could not be computed by Webots, with a null pointer it is easier to detect if the call was invalid or not (see comment on the following line). 
https://github.com/cyberbotics/webots/blob/49573cdfa90e59c7db9f7b3de764ad4d1610c65e/src/controller/c/supervisor.c#L2198
With an array, Webots should send an aswer to the libController and explicitly set the array valued to NaN (same as in the `invalid_vector`)